tasks/controllers/mongodbcontroller.js

```diff
@@ -0,0 +1,80 @@
+const mongoDbController = `import { User } from "../models/user";
+
+export const saveUserController = async (req, res) => {
+  try {
+    const { firstName, lastName, email, password } = req.body;
+    const newUserDetails = { firstName, lastName, email, password };
+    const Newuser = new User(newUserDetails);
+    const data = await Newuser.save();
+    res.status(201).send({
+      message: "User created successfully",
+      data
+    });
+  } catch (err) {
+    res.status(400).send({
+      message: "An error occurred while creating user"
+    });
+  }
+};
```

Comment:
Hi @hezronkimutai, I think we should wrap all async code inside a function that catches the errors instead of having multiple try-catch blocks. This function could be in a helpers file or something, like so

```
exports.catchErrors = (fn) => {
  return function(req, res, next) {
    return fn(req, res, next).catch(next);
  };
};
```

This way we can wrap all async controllers in the routes file like so:
```
router.route('/').post(
    UserMiddleware,
    catchErrors(UserController.createUser)
);
```

Reply:
@its-tayo I totally agree with this suggestion.

Reply:
@its-tayo, this is really nice. I will do that right away.

Reply:
Do you suggest I put that function in the helper or in the middleware folder?

Reply:
@hezronkimutai I think the helper folder is more appropriate.

Reply:
Okey thanks @its-tayo .